docs/elements.md

```diff
@@ -529,13 +556,88 @@ Attribute | Type | Default | Description
 `label` | text | — | A prefix to display before the input box (e.g., `label="$F =$"`).
 `display` | "block" or "inline" | "inline" | How to display the input field.
 `variables` | string | — | A comma-delimited list of symbols that can be used in the symbolic expression.
+`extra-functions` | string | - | A comma-delimited list of extra function names that can be used in the symbolic expression. Note that an exception is raised if the list contains one or more names that coincide with existing reserved functions.
+`enable-function-groups` | string | - | A single-comma-delimited list of extra function group names that can be enabled at runtime. A list of valid function group names includes the following: `trig-base`, `trig-hyperbolic`, `complex-variables`, `logarithmic`, `prog-extended`, `calculus-extended`, or `all` to enable all of the groups (see descriptions below). 
+`enable-constant-groups` | string | - | A single-comma-delimited list of extra symbolic constant group names that can be optionally enabled at runtime. A list of valid constant groups includes the following: `named-constants`, `named-units`, `named-physical-constants`, or `all` to enable all of the groups (see descriptions below).
 `allow-complex` | boolean | false | Whether complex numbers (expressions with `i` or `j` as the imaginary unit) are allowed.
 `imaginary-unit-for-display` | string | `i` | The imaginary unit that is used for display. It must be either `i` or `j`. Again, this is *only* for display. Both `i` and `j` can be used by the student in their submitted answer, when `allow-complex="true"`.
 `allow-blank` | boolean | false |  Whether or not an empty input box is allowed. By default, an empty input box will not be graded (invalid format).
 `blank-value` | string | 0 (zero) | Expression to be used as an answer if the answer is left blank. Only applied if `allow-blank` is `true`. Must follow the same format as an expected user input (e.g., same variables, etc.).
 `size` | integer | 35 | Size of the input box.
 `show-help-text` | boolean | true | Show the question mark at the end of the input displaying required input parameters.
 
+##### Descriptions of ``enable-function-groups`` selections
+
+The next listings document what named function definitions are included
+by enabling these individual groups:
+
+* ``trig-base``: Support for definitions of basic (non-hyperbolic) trigonometric functions
+                      including $cos$, $sin$, $tan$, $sec$, $cot$, $csc$,
+                      ``acos``, ``asin``, ``atan``,
+                      ``asec``, ``acot``, ``acsc``,
+                      ``arccos``, ``arcsin``,
+                      ``arctan``, and ``atan2``.
```

Comment:
A subset of the trig function names are enabled by default. See the file ``lib/python_helper_sympy.py`` (line 32):
```python
self.functions = {
            # These are shown to the student
            'cos'    : sympy.cos,
            'sin'    : sympy.sin,
            'exp'    : sympy.exp,
            'log'    : sympy.log,
            'cos'    : sympy.cos,
            'sin'    : sympy.sin,
            'tan'    : sympy.tan,
            'sec'    : sympy.sec,
            'csc'    : sympy.csc,
            'cot'    : sympy.cot,
            'arccos' : sympy.acos,
            'arcsin' : sympy.asin,
            'arctan' : sympy.atan,
            'atan2'  : sympy.atan2,
            # We need to add in recognition for sympy functions that are involved in constant expressions 
            # that are optionally enabled in the `pl-symbolic-input` element. Otherwise, parsing of these expressions 
            # will fail if they are not listed as known functions at runtime. This is a short list to include by default:
            'sqrt'   : sympy.sqrt,
            'log'    : sympy.log,
            'zeta'   : sympy.zeta,
        }
```